### PR TITLE
Add packages and fix publish script

### LINKED ISSUE #3682: New 0.12 releases are not pushing to crates.io
#### Problem

Solana repo releases after v0.12.0 have not been appearing on crates.io.  Investigate why this upload is failing.

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem

We are failing to publish to crates.io due to script bug and missing some internal dependencies in the CRATES list.

Fixes #3682

### CODE REVIEW

File: ci/publish-crate.sh [hunk 1]
```diff
@@ -19,9 +19,12 @@ CRATES=(
   metrics
   client
   drone
-  programs/{budget_api,config_api,storage_api,token_api,vote_api}
+  instruction-processors/vote_api
```
Comment (mvines) [inline ci/publish-crate.sh]: ugh so sorry for the churn.  instruction-processors are now progams again.

https://media.tenor.com/images/a4ace8088e197c934c8506244a2106cb/tenor.gif

File: ci/publish-crate.sh [hunk 2]
```diff
@@ -19,9 +19,12 @@ CRATES=(
...
   runtime
-  programs/{budget,bpf_loader,config,vote,storage,token,vote}
+  instruction-processors/{budget_api,config_api,storage_api,token_api,vote_signer,exchange_api}
+  instruction-processors/{vote_program,budget_program,bpf_loader,bpf,config_program,exchange_program,failure_program}
+  instruction-processors/{noop_program,stake_api,stake_program,storage_program,token_program}
```
Comment (mvines) [inline ci/publish-crate.sh]: let's group stake_api with the other apis a couple lines up

File: ci/publish-crate.sh [hunk 3]
```diff
@@ -19,9 +19,12 @@ CRATES=(
   metrics
   client
   drone
-  programs/{budget_api,config_api,storage_api,token_api,vote_api}
+  instruction-processors/vote_api
   runtime
-  programs/{budget,bpf_loader,config,vote,storage,token,vote}
+  instruction-processors/{budget_api,config_api,storage_api,token_api,vote_signer,exchange_api}
+  instruction-processors/{vote_program,budget_program,bpf_loader,bpf,config_program,exchange_program,failure_program}
```
Comment (mvines) [inline ci/publish-crate.sh]: bpf_loader is repeated.  I don't think we need to publish `bpf` either

**Comment by danpaul000** [comment]
@mvines I was unable to manually run the above script on my ubuntu VM even with all the env settings from solana-secondary exported to my shell.  Is there some other setting needed?

**Pull request merged**